sonar-php-plugin/src/main/java/org/sonar/plugins/php/AnalysisScanner.java

```diff
@@ -133,12 +137,22 @@ String name() {
 
   @Override
   void scanFile(InputFile inputFile) {
-    if (fileCanBeSkipped(inputFile) && scanFileWithoutParsing(inputFile)) {
+    CpdVisitor cpdVisitor = new CpdVisitor();
```

Comment:
I tried to implement `CpdVisitor` stateless but I filed because tokens can be restored from the cache or computed from Tree. And I would still hide the cache behind `CpdVisitor`.